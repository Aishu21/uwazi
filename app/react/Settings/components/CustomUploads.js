import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import Dropzone from 'react-dropzone';
import PropTypes from 'prop-types';
import React, { Component } from 'react';

import { t } from 'app/I18N';

import { uploadCustom } from '../../Uploads/actions/uploadsActions';

export class CustomUploads extends Component {
  onDrop(files) {
    files.forEach((file) => {
      this.props.upload(file);
    });
  }
  render() {
    return (
      <div className="panel panel-default">
        <div className="panel-heading">{t('System', 'Custom Uploads')}</div>
        <div className="panel-body">
          <Dropzone
            className="upload-box"
            onDrop={this.onDrop.bind(this)}
          >
            <div className="upload-box_wrapper">
              <i className="fa fa-upload" />
              <a className="upload-box_link">Browse files to upload</a>
              <span> or drop your files here.</span>
            </div>
            <div className="protip">
              <i className="fa fa-lightbulb-o" />
            </div>
          </Dropzone>
        </div>
        {this.props.progress && <p>Uploading ...</p>}
        <ul>
          {this.props.customUploads.map((upload) => {
            return <li>{`/uploaded_documents/${upload.get('filename')}`}</li>;
          })}
        </ul>
      </div>
    );
  }
}

CustomUploads.defaultProps = {
  progress: false
};

CustomUploads.propTypes = {
  progress: PropTypes.bool,
  upload: PropTypes.func.isRequired
};

export const mapStateToProps = ({ customUploads, progress }) => ({
    customUploads,
    progress: !!progress.filter((v, key) => key.match(/customUpload/)).size
});

const mapDispatchToProps = dispatch => bindActionCreators({ upload: uploadCustom }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(CustomUploads);
