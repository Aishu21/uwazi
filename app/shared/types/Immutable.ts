/** @format */

export type IImmutable<T> = T extends string
  ? string
  : T extends number
  ? number
  : T extends boolean
  ? boolean
  : T extends null
  ? null
  : T extends undefined
  ? undefined
  : T extends Array<infer Elem>
  ? {
      toJS(): Elem[];
      get(i: number): IImmutable<Elem>;
    }
  : {
      toJS(): T;
      get<Field extends keyof T>(_field: Field): IImmutable<T[Field]>;
    };
