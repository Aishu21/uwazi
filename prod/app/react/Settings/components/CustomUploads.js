"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports.default = exports.mapStateToProps = exports.CustomUploads = void 0;var _redux = require("redux");
var _reactRedux = require("react-redux");
var _reactDropzone = _interopRequireDefault(require("react-dropzone"));
var _immutable = _interopRequireDefault(require("immutable"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));

var _Layout = require("../../Layout");
var _BasicReducer = require("../../BasicReducer");
var _I18N = require("../../I18N");
var _RouteHandler = _interopRequireDefault(require("../../App/RouteHandler"));
var _api = _interopRequireDefault(require("../../utils/api"));
var _UI = require("../../UI");

var _uploadsActions = require("../../Uploads/actions/uploadsActions");function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}var REACT_ELEMENT_TYPE;function _jsx(type, props, key, children) {if (!REACT_ELEMENT_TYPE) {REACT_ELEMENT_TYPE = typeof Symbol === "function" && Symbol["for"] && Symbol["for"]("react.element") || 0xeac7;}var defaultProps = type && type.defaultProps;var childrenLength = arguments.length - 3;if (!props && childrenLength !== 0) {props = { children: void 0 };}if (props && defaultProps) {for (var propName in defaultProps) {if (props[propName] === void 0) {props[propName] = defaultProps[propName];}}} else if (!props) {props = defaultProps || {};}if (childrenLength === 1) {props.children = children;} else if (childrenLength > 1) {var childArray = new Array(childrenLength);for (var i = 0; i < childrenLength; i++) {childArray[i] = arguments[i + 3];}props.children = childArray;}return { $$typeof: REACT_ELEMENT_TYPE, type: type, key: key === undefined ? null : '' + key, ref: null, props: props, _owner: null };}

class CustomUploads extends _RouteHandler.default {
  static async requestState(requestParams) {
    const customUploads = await _api.default.get('customisation/upload', requestParams);
    return [
    _BasicReducer.actions.set('customUploads', customUploads.json)];

  }

  constructor(props, context) {
    super(props, context);
    this.onDrop = this.onDrop.bind(this);
  }

  onDrop(files) {
    files.forEach(file => {
      this.props.upload(file);
    });
  }

  render() {
    return (
      _jsx("div", { className: "panel panel-default" }, void 0,
      _jsx("div", { className: "panel-heading" }, void 0, (0, _I18N.t)('System', 'Custom Uploads')),
      _jsx("div", { className: "panel-body custom-uploads" }, void 0,
      _jsx(_reactDropzone.default, {
        className: "upload-box",
        onDrop: this.onDrop }, void 0,

      _jsx("div", { className: "upload-box_wrapper" }, void 0,
      _jsx(_UI.Icon, { icon: "upload" }),
      _jsx("button", { className: "upload-box_link", type: "button" }, void 0, "Browse files to upload"),
      _jsx("span", {}, void 0, " or drop your files here.")),

      this.props.progress && _jsx("div", { className: "uploading" }, void 0, _jsx(_UI.Icon, { icon: "spinner", spin: true }), " Uploading ...")),

      _jsx("ul", {}, void 0,
      this.props.customUploads.map((upload) =>
      _jsx("li", {}, upload.get('filename'),
      _jsx(_Layout.Thumbnail, { file: `/assets/${upload.get('filename')}` }),
      _jsx("div", { className: "info" }, void 0, "URL:",
      _jsx("br", {}),
      _jsx("span", { className: "thumbnail-url" }, void 0, `/assets/${upload.get('filename')}`),
      _jsx(_Layout.ConfirmButton, { action: () => this.props.deleteCustomUpload(upload.get('_id')) }, void 0, "Delete"))))))));







  }}exports.CustomUploads = CustomUploads;


CustomUploads.defaultProps = {
  progress: false };


CustomUploads.propTypes = {
  progress: _propTypes.default.bool,
  customUploads: _propTypes.default.instanceOf(_immutable.default.List).isRequired,
  upload: _propTypes.default.func.isRequired,
  deleteCustomUpload: _propTypes.default.func.isRequired };


const mapStateToProps = ({ customUploads, progress }) => ({
  customUploads,
  progress: !!progress.filter((v, key) => key.match(/customUpload/)).size });exports.mapStateToProps = mapStateToProps;


const mapDispatchToProps = dispatch => (0, _redux.bindActionCreators)({ upload: _uploadsActions.uploadCustom, deleteCustomUpload: _uploadsActions.deleteCustomUpload }, dispatch);var _default =

(0, _reactRedux.connect)(mapStateToProps, mapDispatchToProps)(CustomUploads);exports.default = _default;