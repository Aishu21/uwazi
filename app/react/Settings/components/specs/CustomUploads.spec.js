import Immutable from 'immutable';
import React from 'react';

import { shallow } from 'enzyme';

import { CustomUploads, mapStateToProps } from '../CustomUploads';

describe('CustomUploads', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      upload: jasmine.createSpy('upload')
    };
  });

  const render = () => {
    component = shallow(<CustomUploads {...props}/>);
  };

  it('should render CustomUploads component with uploaded files', () => {
    props.customUploads = Immutable.fromJS([
      { filename: 'file1' },
      { filename: 'file2' },
    ]);
    render();
    expect(component).toMatchSnapshot();
  });

  describe('when upload on progress', () => {
    it('should render on progress feedback', () => {
      props.progress = true;
      render();
      expect(component).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('should map current progress and files to props', () => {
      const state = {
        customUploads: 'customUploads',
        progress: Immutable.fromJS({})
      };

      let props = mapStateToProps(state);
      expect(props.customUploads).toBe('customUploads');
      expect(props.progress).toBe(false);

      state.progress = Immutable.fromJS({ customUpload_unique_id: 1, customUpload_unique_id2: 100 });
      props = mapStateToProps(state);
      expect(props.progress).toBe(true);

      state.progress = Immutable.fromJS({ not_custom_upload: 9 });
      props = mapStateToProps(state);
      expect(props.progress).toBe(false);
    });
  });
});
