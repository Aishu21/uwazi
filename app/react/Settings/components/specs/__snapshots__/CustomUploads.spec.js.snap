// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomUploads should render CustomUploads component 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    Custom Uploads
  </div>
  <div
    className="panel-body"
  >
    <t
      className="upload-box"
      disableClick={false}
      disablePreview={false}
      disabled={false}
      maxSize={Infinity}
      minSize={0}
      multiple={true}
      onDrop={[Function]}
      preventDropOnDocument={true}
    >
      <div
        className="upload-box_wrapper"
      >
        <i
          className="fa fa-upload"
        />
        <a
          className="upload-box_link"
        >
          Browse files to upload
        </a>
        <span>
           or drop your files here.
        </span>
      </div>
      <div
        className="protip"
      >
        <i
          className="fa fa-lightbulb-o"
        />
      </div>
    </t>
  </div>
</div>
`;

exports[`CustomUploads should render CustomUploads component with uploaded files 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    Custom Uploads
  </div>
  <div
    className="panel-body"
  >
    <t
      className="upload-box"
      disableClick={false}
      disablePreview={false}
      disabled={false}
      maxSize={Infinity}
      minSize={0}
      multiple={true}
      onDrop={[Function]}
      preventDropOnDocument={true}
    >
      <div
        className="upload-box_wrapper"
      >
        <i
          className="fa fa-upload"
        />
        <a
          className="upload-box_link"
        >
          Browse files to upload
        </a>
        <span>
           or drop your files here.
        </span>
      </div>
      <div
        className="protip"
      >
        <i
          className="fa fa-lightbulb-o"
        />
      </div>
    </t>
  </div>
</div>
`;

exports[`CustomUploads when upload on progress should render on progress feedback 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    Custom Uploads
  </div>
  <div
    className="panel-body"
  >
    <t
      className="upload-box"
      disableClick={false}
      disablePreview={false}
      disabled={false}
      maxSize={Infinity}
      minSize={0}
      multiple={true}
      onDrop={[Function]}
      preventDropOnDocument={true}
    >
      <div
        className="upload-box_wrapper"
      >
        <i
          className="fa fa-upload"
        />
        <a
          className="upload-box_link"
        >
          Browse files to upload
        </a>
        <span>
           or drop your files here.
        </span>
      </div>
      <div
        className="protip"
      >
        <i
          className="fa fa-lightbulb-o"
        />
      </div>
    </t>
  </div>
  <p>
    Uploading ...
  </p>
</div>
`;
